optimization debian package manager tweaks

rajpratik71 opened the pull request:
By default, Ubuntu or Debian based "apt" or "apt-get" system installs recommended but not suggested packages .

By passing "--no-install-recommends" option, the user lets apt-get know not to consider recommended packages as a dependency to install.

This results in smaller downloads and installation of packages .

Refer to blog at [Ubuntu Blog](https://ubuntu.com/blog/we-reduced-our-docker-images-by-60-with-no-install-recommends) .

Signed-off-by: Pratik Raj <rajpratik71@gmail.com>

jerroydmoore commented:
@Rajpratik71,

This is a great contribution! Have you measured how much smaller the new images are using the `--no-install-recommends` flag?

rajpratik71 committed:
Some debian package manager tweaks

By default, Ubuntu or Debian based "apt" or "apt-get" system installs recommended but not suggested packages . 

By passing "--no-install-recommends" option, the user lets apt-get know not to consider recommended packages as a dependency to install.

This results in smaller downloads and installation of packages .

Refer to blog at [Ubuntu Blog](https://ubuntu.com/blog/we-reduced-our-docker-images-by-60-with-no-install-recommends) .

Signed-off-by: Pratik Raj <rajpratik71@gmail.com>

rajpratik71 commented:
> @Rajpratik71,
> 
> This is a great contribution! Have you measured how much smaller the new images are using the `--no-install-recommends` flag?

didn't done that research in terms of figure but in general the improvement is acceptable
